gc/base/ParallelDispatcher.cpp

```diff
@@ -630,3 +644,140 @@ MM_ParallelDispatcher::reinitAfterFork(MM_EnvironmentBase *env, uintptr_t newThr
 
 	startUpThreads();
 }
+
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+/**
+ * Contract the thread pool by shutting down threads in pool to obtain newThreadCount.
+ *
+ * This API assumes  CRIU is the only consumer (not tested for general use).
+ * The following conditions are assumed while contracting the thread pool:
+ *     1) API can only be called once during VM lifetime
+ *     2) NOT holding exclusive VM access
+ *     3) Dispatcher is idle (no task can be dispatched)
+ *     4) Can't externally enter shutdown from another source
+ */
+void
+MM_ParallelDispatcher::contractThreadPool(MM_EnvironmentBase *env, uintptr_t newThreadCount)
+{
+	Assert_MM_false(_workerThreadsReservedForGC);
+	Assert_MM_false(_inShutdown);
```

Comment:
do we end up calling shutDownThreads after this?
either way, we should probably assert the same at the start of shutDownThreads

Reply:
shutDownThreads should only be called during VM shutdown after this point